### PR TITLE
Walletconnect support for Unlock account

### PR DESCRIPTION
**Pull request by julien51**
# Description

By adding support fo WalletConnect we make it possible for Unlock Account users to authenticate thru WalletConnect when user 3rd party applications.
We will probably need to work a lot on the UI but that is the beginning!

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
One thing we must do is simplify the auth flow so that only users with Unlock accounts are able to login (even though strangely it works with other users!).
Our redirect scheme seems to break things when the `uri` is passed as query string.

**Comment by julien51** [comment]
This is a first start. The goal is to eventually submit this as an app to WalletConnets' registry so that users can select their Unlock Account on the WalletConnect modal.

**Review by searchableguy — APPROVED** [review]
LGTM for a start.

File: unlock-app/src/hooks/useWalletConnectClient.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        })
+      }
+      if (payload.method === 'wallet_switchEthereumChain') {
+        return walletConnect.approveRequest({
+          id: payload.id,
+          jsonrpc: '2.0',
+          result: null,
+        })
+      }
+      // Ignore other types
```
[the remark this answers is not in the document]
Comment (julien51) [reply unlock-app/src/hooks/useWalletConnectClient.ts:66]: Good point. I'll add that in the next change.

**Pull request merged**